hbase-server/src/main/java/org/apache/hadoop/hbase/util/RegionMover.java

```diff
@@ -755,7 +755,8 @@ private ServerName getServerNameForRegion(RegionInfo region) throws IOException
       return null;
     }
     HRegionLocation loc =
-      conn.getRegionLocator(region.getTable()).getRegionLocation(region.getStartKey(), true);
+      conn.getRegionLocator(region.getTable()).getRegionLocation(region.getStartKey(),
```

Comment:
Is there already a test, which covers the old as well as the new behavior?

Reply:
Existing test cases are good. But yes good to add additional test case specific for replicaId. @BukrosSzabolcs can you add a test case in TestRegionMover ?

Reply:
The issue was found by a test. IntegrationTestTimeBoundedRequestsWithRegionReplicas failed while running with chaos monkey using GracefulRollingRestartRsAction. So there is coverage just not in TestRegionMover. Would that be sufficient?

Reply:
It's great that IT test found this issue but they are not run as part of QA build. Hence, if we can have one unit test, that would be really great. QA will always run it. Sounds good @BukrosSzabolcs ?

Reply:
However, since we have an IT that covers this, ideally it should be fine and maybe you can take up unit test as separate child Jira of this Jira.
Should be good @HorizonNet ?

Reply:
I realized my original description of the issue is not correct. This does not prevent the moving of the region, the result of getServerNameForRegion() is only used in the validation of the move, so it only forces the move to try to repeat itself because it does not realize the move already happened. So it just slows down the process but not break it.

I tried to write a test for this, but the only thing I could come up with is a test that goes trough regions on a rs and checks if getServerNameForRegion founds the correct server for it. I can add it if you insist but I see very little added value.

Reply:
Then let it be.